plugins/toolbox/src/components/Converters/SLACalculator.tsx

```diff
@@ -0,0 +1,122 @@
+import { useStyles } from '../../utils/hooks';
+import React from 'react';
+import {
+  Divider,
+  FormControl,
+  Grid,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import { PasteFromClipboardButton } from '../Buttons/PasteFromClipboardButton';
+import { ClearValueButton } from '../Buttons/ClearValueButton';
+import { CopyToClipboardButton } from '../Buttons/CopyToClipboardButton';
+
+export const SLACalculator = () => {
+  const styles = useStyles();
+  const [input, setInput] = React.useState('');
+  const [output, setOutput] = React.useState({
+    daily: '',
+    weekly: '',
+    monthly: '',
+    quarterly: '',
+    yearly: '',
+  });
+
+  const convertTime = (value: number) => {
+    let minutes = Math.floor(value / 60);
+    const seconds = Math.floor(value % 60);
+    if (minutes >= 60) {
+      const hours = Math.floor(minutes / 60);
+      minutes = minutes % 60;
+      return `${hours}h ${minutes}m ${seconds}s`;
+    }
+
+    return `${minutes}m ${seconds}s`;
+  };
+
+  const handleChange = (value: string) => {
+    if (value.length === 0) {
+      setInput('');
+      setOutput({
+        daily: '',
+        weekly: '',
+        monthly: '',
+        quarterly: '',
+        yearly: '',
+      });
+      return;
+    }
+
+    const base = parseFloat(value);
+    if (isNaN(base)) {
+      return;
+    }
+    if (base >= 100) {
```

Comment:
How about values under zero? Instead of returning, I think you could just force the min, and max values and show it to the user. For example, if the user inputs 101 it will correct it to 100. Should be easy to do here with just a few Math.min, Math.max calls.

Reply:
Don't know what you mean about using Math.min, Math.max calls here - added if for max value and correctness logic. Below zero will not be allowed by the `isValidFloat` function. If you feel like `Math.max` will be better - could you please provide an example?